packages/gatsby-plugin-cms/src/gatsby-node.ts

```diff
@@ -151,19 +160,24 @@ export const createPages = async ({ graphql, reporter }: CreatePagesArgs) => {
     extensions: ['.ts'],
     presets: ['@babel/preset-typescript'],
   })
-  const { contentTypes } = require(SHADOWED_INDEX_PATH) as {
-    schemas: Schemas
-    contentTypes: ContentTypes
+  const {
+    builderConfig: {
+      contentTypes: ctypes = {},
+      blocks: blks = {},
+      messages = {},
+    } = {},
+  } = require(SHADOWED_INDEX_PATH) as {
+    builderConfig: BuilderConfig
   }
 
-  // Transform all contentTypes into CMS ready contentType json
-  const contentTypesCMS = Object.keys(contentTypes).reduce((acc, key) => {
-    const ct = contentTypes[key]
+  const blocks = Object.keys(blks).map((k) => ({
+    name: k,
+    schema: blks[k],
```

Comment:
I'd rather not use cryptic names.

`blks` could be `blockNames` and `k` could be `blockName`